atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -51,49 +44,89 @@ class SendMailPlugin(ExitPlugin):
     # symbolic constants for states
     MANUAL_SUCCESS = 'manual_success'
     MANUAL_FAIL = 'manual_fail'
+    MANUAL_CANCELED = 'manual_canceled'
     AUTO_SUCCESS = 'auto_success'
     AUTO_FAIL = 'auto_fail'
     AUTO_CANCELED = 'auto_canceled'
-
-    allowed_states = set([MANUAL_SUCCESS, MANUAL_FAIL, AUTO_SUCCESS, AUTO_FAIL, AUTO_CANCELED])
-
-    PDC_TOKEN_FILE = 'pdc.token'
-    PDC_CONTACT_ROLE = 'Devel_Owner'
-
-    def __init__(self, tasker, workflow, send_on=None, url=None, submitter='unknown', pdc_url=None,
-                 pdc_verify_cert=True, pdc_component_df_label="com.redhat.component", pdc_secret_path=None,
-                 pdc_contact_role=None, smtp_uri=None, from_address=None,
-                 error_addresses=None):
+    DEFAULT_SUBMITTER = 'Unknown'
+
+    allowed_states = set([MANUAL_SUCCESS, MANUAL_FAIL, MANUAL_CANCELED,
+                          AUTO_SUCCESS, AUTO_FAIL, AUTO_CANCELED])
+
+    def __init__(self, tasker, workflow,
+                 send_on=None,
+                 url=None,
+                 smtp_host=None,
```

Comment:
My preference would be to move this parameter to before the optional ones, i.e. right after tasker and workflow. We have this plugin marked as "not yet enabled" in the docs so it's safe to rearrange parameters (which we're doing anyway in this PR).

Another option: verify the parameter is not set and raise an exception if it is not.